### ISSUE TITLE
Explore: For stateless Grafana, the shortlink is not persisited

### ISSUE THREAD

**Issue by ivanahuckova** [grafana/grafana#77746]
It would be worth to explore if we know when Grafana is stateless and in those cases we wouldn't offer shortlinks as they will be deleted. Maybe there is some config/other way to learn about stateless Grafana.

During user interviews, we've learned that some of the internal users don't use shortlink as sometimes it does not work. To build a confidence in the feature, we should only offer it when we know it will be persisted.

**Comment by gelicia** [comment]
We're actually just reworking this menu in general. I can add this requirement to #59370

**Comment by gelicia** [comment]
I asked the backend folks what the best things to check are to know this data wont persist long and shouldn't be saved. As of now, if this can't be answered, its actually probably best if we wait.

**Comment by ifrost** [comment]
Is the link disabled in dashboards when Grafana is stateless?

**Comment by ivanahuckova** [comment]
> Is the link disabled in dashboards when Grafana is stateless?

No. And as @gelicia mentioned, I am not sure that you can easily know if Grafana is stateless or not. So this would require to figure out how/if we can check if Grafana is stateless (maybe adding a config for it?) and then just using it.

**Comment by ifrost** [comment]
It seems like it depends on how a specific Grafana instance is deployed and managed which I think is not something that Grafana app would be aware of in any way. It may happen in cases when Grafana database is provisioned and the instance is frequently restarted with database being removed on each restart.

In theory we could provide some config option to hide the shortlink option. We could also consider having more generic config that would hide various options that cannot be use as-code and the intention of the admin is to clear the database frequently.

**Comment by samjewell** [comment]
![image](https://github.com/grafana/grafana/assets/2903904/adff74f7-fec7-41cf-a88e-5960b1a5b7be)

Should this also apply to "Copy shortlink" on a single log-line? I found myself doing this today, but fully expecting it to break within hours.

**Comment by gtk-grafana** [comment]
Bump: we had a user report this as a paper-cut in their logs UX. Might be nice to add a config that users can set so we can avoid exposing urls in the DB that will get wiped.

**Comment by gelicia** [comment]
I created #83425 to track the setting that would need to be added to enable this.
